.github/workflows/gotest.yml

```diff
@@ -59,8 +65,38 @@ jobs:
           mv go.mod.bak go.mod
           mv go.sum.bak go.sum
         working-directory: docs/examples/kubo-as-a-library
-      - uses: actions/upload-artifact@v3
+      - name: Create a proper JUnit XML report
+        uses: pl-strflt/gotest-json-to-junit-xml@v1
+        with:
+          input: test/unit/gotest.json
+          output: test/unit/gotest.junit.xml
+        if: failure() || success()
+      - name: Archive the JUnit XML report
+        uses: actions/upload-artifact@v3
         with:
           name: unit
           path: test/unit/gotest.junit.xml
-        if: always()
+        if: failure() || success()
+      - name: Create a HTML report
+        uses: pl-strflt/junit-xml-to-html@v1
+        with:
+          mode: no-frames
+          input: test/unit/gotest.junit.xml
+          output: test/unit/gotest.html
+        if: failure() || success()
+      - name: Archive the HTML report
+        uses: actions/upload-artifact@v3
+        with:
+          name: html
+          path: test/unit/gotest.html
+        if: failure() || success()
+      - name: Create a Markdown report
+        uses: pl-strflt/junit-xml-to-html@v1
+        with:
+          mode: summary
+          input: test/unit/gotest.junit.xml
+          output: test/unit/gotest.md
+        if: failure() || success()
+      - name: Set the summary
+        run: cat test/unit/gotest.md >> $GITHUB_STEP_SUMMARY
+        if: failure() || success()
```

Comment:
💡 We could also post a link to the summary in a PR when the tests are failing. Let me know if that's something you wanted.

Reply:
Personally I'd rather skip this, as it would leave behind a bunch of noise in the PR from failed test runs.

Reply:
We could address this by using a sticky comment i.e. one comment per PR with the test results. But I think it's fair to start without it. We can always revisit this in the future. Just wanted to throw it out there as an idea.